bot/cogs/utils.py

```diff
@@ -261,7 +240,54 @@ async def send_pep_zero(self, ctx: Context) -> None:
         pep_embed.add_field(name="Created", value="13-Jul-2000")
         pep_embed.add_field(name="Type", value="Informational")
 
-        await ctx.send(embed=pep_embed)
+        return pep_embed
+
+    @async_cache(arg_offset=1)
+    async def get_pep_embed(self, pep_nr: int, ctx: Context) -> Optional[Embed]:
+        """Fetch, generate and return PEP embed."""
+        if pep_nr not in self.peps:
+            log.trace(f"PEP {pep_nr} was not found")
+            not_found = f"PEP {pep_nr} does not exist."
+            await ctx.send(
+                embed=Embed(title="PEP not found", description=not_found, colour=Colour.red())
+            )
+            return
+        response = await self.bot.http_session.get(self.peps[pep_nr])
+
+        if response.status == 200:
+            log.trace(f"PEP {pep_nr} found")
+            pep_content = await response.text()
+
+            # Taken from https://github.com/python/peps/blob/master/pep0/pep.py#L179
+            pep_header = HeaderParser().parse(StringIO(pep_content))
+
+            # Assemble the embed
+            pep_embed = Embed(
+                title=f"**PEP {pep_nr} - {pep_header['Title']}**",
+                description=f"[Link]({self.base_pep_url}{pep_nr:04})",
+            )
+
+            pep_embed.set_thumbnail(url=ICON_URL)
+
+            # Add the interesting information
+            fields_to_check = ("Status", "Python-Version", "Created", "Type")
+            for field in fields_to_check:
+                # Check for a PEP metadata field that is present but has an empty value
+                # embed field values can't contain an empty string
+                if pep_header.get(field, ""):
+                    pep_embed.add_field(name=field, value=pep_header[field])
+            return pep_embed
+        else:
+            log.trace(
+                f"The user requested PEP {pep_nr}, but the response had an unexpected status code: "
+                f"{response.status}.\n{response.text}"
+            )
+
+            error_message = "Unexpected HTTP error during PEP search. Please let us know."
+            await ctx.send(
+                embed=Embed(title="Unexpected error", description=error_message, colour=Colour.red())
+            )
```

Comment:
Bit nicer to avoid putting args on a new line when possible IMO.
```suggestion
            embed = Embed(title="Unexpected error", description=error_message, colour=Colour.red())
            await ctx.send(embed=embed)
```